interview-questions.md

```diff
@@ -7,44 +7,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: The foreign key is added to the database table that doesn't have the primary key. In this example, The model cohort that has many students  would  be the table in which you would add the foreign key the relationship of one to many requires a primary key which would then be used to identify each student on any other table in the Student model database cohort. The rails application automatically generates the primary key so to add a foreign key you would simply have to figure out which columns on your new table corresponds with the primary key on the main table and link the two tables. The student table will be your primary table on which all the other relationships will exist it will contain the primary key. The cohort table will contain the foreign key so you will have to add this key to the cohort table. The cohort table contains only those students that are in a particular cohort not all the students like the student table does.
 
-  Researched answer:
+  Researched answer: Rails is much smarter than I realized you just have to use a migration file to validate the foreign key. You should not use add_foreign_key or add_concurrent_foreign_key more than once per migration file unless tables are identical. Rails includes helpers with migration to ensure that foreign key restraints are added to each database table. Rails won't let you delete foreign keys with restraints. You can use a plug-in that does this for you. 
 
 
 
 2. Which RESTful routes must always be passed params? Why?
 
```

Comment:
Look at the restful route diagram. There should be a column designated for the path. Do you see the routes that require an id as param now?